Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2678 @@
+
+- [Purpose Strings in the Info.plist File](#Purpose-Strings-in-the-Info.plist-File)
+- [Code Signing Entitlements File](#Code-Signing-Entitlements-File)
+- [Embedded Provisioning Profile File](#Embedded-Provisioning-Profile-File)
+- [Entitlements Embedded in the Compiled App Binary](#Entitlements-Embedded-in-the-Compiled-App-Binary)
+- [Source Code Inspection](#Source-Code-Inspection)
+
+##### Purpose Strings in the Info.plist File
+
+An app's [`Info.plist`](https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/ExpectedAppBehaviors/ExpectedAppBehaviors.html#//apple_ref/doc/uid/TP40007072-CH3-SW5) or *information property list* file contains, among others, the app's overview of protected data and resources as a set of key-value pairs. Each value contains the so-called *purpose string* or *usage description string* (mandatory from iOS 10 onward), which is a custom text that is offered to users in the system's permission request alert when requesting permission to access protected data or resources.
+
+If having the original source code, you can verify the permissions included in the `Info.plist` file:
+
+- open the project with Xcode
+- find and open the `Info.plist` file in the default editor and search for the keys starting with `"Privacy -"`
+- you may switch the view to display the raw values by right-clicking and selecting "Show Raw Keys/Values" (this way for example `"Privacy - Location When In Use Usage Description"` will turn into `NSLocationWhenInUseUsageDescription`)
+
+If only having the IPA:
+
+- unzip the IPA
+- the `Info.plist` is located in `Payload/<appname>.app/Info.plist`
+- if the file comes in binary format (bplist), convert it to XML format
+
+    On macOS:
+
+    ```bash
+    $ plutil -convert xml1 Info.plist
+    ```
+
+    On Linux:
+
+    ```bash
+    $ apt install libplist-utils
+    $ plistutil -i Info.plist -o Info_xml.plist
+    ```
+- inspect all *purpose strings Info.plist keys*, usually ending with `UsageDescription`
+
+For an overview of the different *purpose strings Info.plist keys* available see Table 1-2 at the [Apple App Programming Guide for iOS](https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/ExpectedAppBehaviors/ExpectedAppBehaviors.html#//apple_ref/doc/uid/TP40007072-CH3-SW7 "Apple App Programming Guide for iOS"). Click on the provided links to see the full description of each key in the [CocoaKeys reference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html). If the app attempts to gain access without having provided a corresponding purpose string, the app will exit.
+
+Following these guidelines should make it relatively simple to evaluate each and every entry in the `Info.plist` file to check if the permission makes sense.
+
+For example, the following lines were extracted from a `Info.plist` file used by a Solitaire game:
```

Comment:
Is this now a "bad example" or was this app actually in the app store?

Reply:
this should be a "bad example" regarding permissions. It was already there before my changes but it does not have any references, maybe we can change the sentence to something else: "imagine that the following lines.." or "If you would find the following lines...in a Solitaire game"

Reply:
Yeah, would be better with "For example, imagine the following lines...". Otherwise the statement is a bit misleading.

Reply:
nice, done

Reply:
done